src/HasMedia/HasMediaTrait.php

```diff
@@ -130,14 +130,20 @@ public function addMediaFromUrl(string $url, ...$allowedMimeTypes)
 
         $filename = basename(parse_url($url, PHP_URL_PATH));
 
-        $mediaExt = explode('/', mime_content_type($tmpFile));
-
         if ($filename === '') {
             $filename = 'file';
         }
 
+        $mediaExt = explode('/', mime_content_type($tmpFile));
+
+        $extension = $mediaExt[1] ?? null;
+
+        if (! $extension) {
+            throw MimeTypeNotAllowed::create($filename, $allowedMimeTypes);
```

Comment:
This doesn't seem right. Files without extensions are allowed.  The mimetype is already checked on line 129.

Reply:
That is true, I'll remove the extra check